app/components/post/post_message.js

```diff
@@ -0,0 +1,248 @@
+        }
+
+        return null;
+    };
+
+    render() {
+        const {
+            canDelete,
+            canEdit,
+            hasBeenDeleted,
+            isFlagged,
+            isPendingOrFailedPost,
+            isSystemMessage,
+            intl,
+            flagPost,
+            message,
+            navigator,
+            onFailedPostPress,
+            onPostDelete,
+            onPostEdit,
+            onPress,
+            renderReplyBar,
+            showReplyBar,
+            theme,
+            toggleSelected,
+            unflagPost
+        } = this.props;
+        const {formatMessage} = intl;
+        const actions = [];
+        const style = getStyleSheet(theme);
+        const blockStyles = getMarkdownBlockStyles(theme);
+        const textStyles = getMarkdownTextStyles(theme);
+        const messageStyle = isSystemMessage ? [style.message, style.systemMessage] : style.message;
+
+        // we should check for the user roles and permissions
+        if (!isPendingOrFailedPost) {
+            if (isFlagged) {
+                actions.push({
+                    text: formatMessage({id: 'post_info.mobile.unflag', defaultMessage: 'Unflag'}),
+                    onPress: unflagPost
+                });
+            } else {
+                actions.push({
+                    text: formatMessage({id: 'post_info.mobile.flag', defaultMessage: 'Flag'}),
+                    onPress: flagPost
+                });
+            }
+
+            if (canEdit) {
+                actions.push({text: formatMessage({id: 'post_info.edit', defaultMessage: 'Edit'}), onPress: onPostEdit});
+            }
+
+            if (canDelete && !hasBeenDeleted) {
+                actions.push({text: formatMessage({id: 'post_info.del', defaultMessage: 'Delete'}), onPress: onPostDelete});
+            }
+        }
+
+        let messageComponent;
+        if (hasBeenDeleted) {
+            messageComponent = (
+                <FormattedText
+                    style={messageStyle}
+                    id='post_body.deleted'
+                    defaultMessage='(message deleted)'
+                />
+            );
+        } else if (message.length) {
+            messageComponent = (
+                <View style={{flexDirection: 'row'}}>
+                    <View style={[{flex: 1}, (isPendingOrFailedPost && style.pendingPost)]}>
+                        <Markdown
+                            baseTextStyle={messageStyle}
+                            textStyles={textStyles}
+                            blockStyles={blockStyles}
+                            value={message}
+                            onLongPress={this.showOptionsContext}
+                            navigator={navigator}
+                        />
+                    </View>
+                </View>
+            );
+        }
+
+        return (
+            <View style={style.messageContainerWithReplyBar}>
+                {showReplyBar && renderReplyBar()}
```

Comment:
Do we need to pass in both of these? Maybe `renderReplyBar` could just check whatever would be passed as `showReplyBar` itself